fe/fe-core/src/main/java/com/starrocks/catalog/ResourceMgr.java

```diff
@@ -106,19 +119,71 @@ private void onDropResource(Resource resource) {
     }
 
     public boolean containsResource(String name) {
-        return nameToResource.containsKey(name);
+        rwlock.readLock().lock();
+        try {
+            return nameToResource.containsKey(name);
+        } finally {
+            rwlock.readLock().unlock();
+        }
     }
 
     public Resource getResource(String name) {
-        return nameToResource.get(name);
+        rwlock.readLock().lock();
+        try {
+            return nameToResource.get(name);
+        } finally {
+            rwlock.readLock().unlock();
+        }
+    }
+
+    /**
+     * alter resource statement only support external hive now .
+     * @param stmt
+     * @throws DdlException
+     */
+    public void alterResource(AlterResourceStmt stmt) throws DdlException {
+        rwlock.writeLock().lock();
+        try {
+            // check if the target resource exists .
+            String name = stmt.getResourceName();
+            Resource resource = this.getResource(name);
+            if (resource == null) {
+                throw new DdlException("Resource(" + name + ") does not exist");
+            }
+
+            if (resource instanceof HiveResource) {
+                // 1. alter the resource properties
+                // 2. update the nameToResource
+                // 3. drop the cache
+                // 4. update the edit log
+                ((HiveResource) resource).alterProperties(stmt.getProperties());
+                nameToResource.put(name, resource);
+                Catalog.getCurrentCatalog().getHiveRepository().clearCache(resource.getName());
```

Comment:
Why clear cache

Reply:
Make sure that the HiveMetaClient object is freshest .
HiveMetaClient  is  initialized by resource properties  metastoreURIs ,when we alter resource properties , we make sure that the cache HiveMetaClient  object can be refreshed and the metastore uri is freshest too .